misc/_common.py

```diff
@@ -0,0 +1,89 @@
+from arcgis.gis import GIS
+import datetime
+
+"""accounts to keep from user/groups/items deletion"""
+ignore_accounts = ['andrew', 'andrew.chapkowski', 'dvitale', 'david.vitale',
+                   'atma.mani', 'john.yaist', 'bill.major', 'YJiang',
+                   'rohit.singh', 'rohitgeo', 'gbochenek_python',
+                   'system_publisher', 'admin', 'portaladmin',
+                   'Demo_User', 'First_User', 'Second_User',
+                   'api_data_owner', 'arcgis_python', 'temp_execution']
+
+"""accounts that you want to delete groups and items, but keep user"""
+target_accounts = ['arcgis_python']
+
+"""create GIS connection via admin credentials"""
+# gis = GIS(profile='your_portal_profile', verify_cert=False)
+gis = GIS("https://pythonapi.playground.esri.com/portal","temp_execution", "temp_execution123")
+
+
+def delete_items(user):
+    """deletes the user items"""
+    folders = [None] + user.folders
+    for folder in folders:
+        print("=== deleting inside folder: %s" % folder)
+        for item in user.items(folder=folder, max_items=255):
+            print("=== deleting the item: %s" % item.id)
+            if item.protected:
+                item.protect(False)
+            depending_items = item.dependent_to()
+            if not depending_items['list']:
```

Comment:
Can we add logic to loop through each of the depending item and delete that before deleting the current item? Also, can you embed the delete call within try/except block?